tensorboard/components/vz_line_chart2/microbenchmark/renders_spec.ts

```diff
@@ -0,0 +1,372 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {benchmark, Size} from './spec.js';
+import {polymerFlush} from './polymer_util.js';
+
+const DATA_POINTS = {
+  sine1k: [...new Array(1000)].map((_, index) => ({
+    step: index,
+    wall_time: index * 1000,
+    scalar: Math.sin(index / (2 * Math.PI)),
+  })),
+  cosine1k: [...new Array(1000)].map((_, index) => ({
+    step: index,
+    wall_time: index * 1000,
+    scalar: Math.cos(index / (2 * Math.PI)),
+  })),
+  cosine100k: [...new Array(100000)].map((_, index) => ({
+    step: index,
+    wall_time: index * 1000,
+    scalar: Math.cos(index / (2 * Math.PI)),
+  })),
+};
+
+const FIVE_HUNDRED_1K_DATA_POINTS = [...new Array(500)].map((_, index) => {
+  const paddedIndex = (String(index) as any).padStart(5, '0');
+  const name = `p${paddedIndex};`;
+  return {
+    name,
+    data: [...new Array(1000)].map((_, index) => ({
+      step: index,
+      wall_time: index * 1000,
+      scalar: Math.sin(index / (2 * Math.PI)) + Math.random() - 0.5,
+    })),
+  };
+});
+
+benchmark({
+  name: 'charts init',
+
+  size: Size.LARGE,
+
+  async run(context) {
+    context.chart = document.createElement('vz-line-chart2') as any;
+    context.chart.style.height = '100%';
+    context.container.appendChild(context.chart);
+
+    context.chart.setVisibleSeries([]);
+
+    await polymerFlush();
+  },
+
+  afterEach(context) {
+    context.container.removeChild(context.chart);
+  },
+});
+
+benchmark({
+  name: 'charts init + 1k point draw',
+
+  size: Size.LARGE,
+
+  async run(context) {
+    context.chart = document.createElement('vz-line-chart2') as any;
+    context.chart.style.height = '100%';
+    context.container.appendChild(context.chart);
+
+    context.chart.setSeriesData('sine', DATA_POINTS.sine1k);
+    context.chart.setVisibleSeries(['sine']);
```

Comment:
Just an observation or discussion point: In the ML microbenchmarks I've worked on or seen, there is usually a couple of "warm-up" calls that are not taken into account for timing purpose. This is to cover the very common property in which the first call is slower than subsequent ones due to the initial setting up of some states. 

This is not the case here. This probably makes sense because in UI performance, the first-render performance does matter. I'd like to hear your thoughts if you have any.